controller/system_rollout_controller.go

```diff
@@ -427,6 +427,8 @@ func (c *SystemRolloutController) systemRollout() error {
 
 	case longhorn.SystemRestoreStateRestoring:
 		c.restore(types.APIExtensionsKindCustomResourceDefinitionList, c.restoreCustomResourceDefinitions, log)
+		c.restore(types.KubernetesKindDaemonSetList, c.restoreDaemonSets, log)
```

Comment:
Do you mean we need to restore longhorn-manager DS and engine image first before other resources?

note: The resource restore order should be the same as installation.

Reply:
After discussing with @c3y1huang , the mandatory DSs need to start first before other resources to avoid this kind of error like engine binary is not ready on nodes. Those are including longhorn manager, engine image, and CSI plugin.